include/envoy/router/router.h

```diff
@@ -216,14 +216,23 @@ class HashPolicy {
   virtual ~HashPolicy() {}
 
   /**
+   * A callback used for requesting that a random cookie be set with the given
```

Comment:
I don't think we really want a random cookie.

Imagine we're doing this because we want a client to consistently land on the same server.  now imagine the client is using H2, and sends 3 requests "simultaneously" on one connection.  Each will have no set cookie, so each will get a random hash, land on a random backend, and get a random cookie set. Whichever response lands last will "win".  We probably want to hash on something stable for the connection to ensure we get consistent routing before the cookie is  received.

Reply:
Also an integration test for this case would be nice, making sure the cookie is consistent (don't care what it is) between requests.   You should be able to crib heavily from (or simply extend config for) existing http2 tests which have multiple simultaneous requests and responses.

Reply:
Should this comment ("random") be updated?